### PR TITLE
Fixing py-solc-x usage to automatically download or compile correct solc binary and use it

### PR DESCRIPTION
**Pull request by jimtje**
#1424

### CODE REVIEW

File: mythril/ethereum/util.py [hunk 1]
```diff
@@ -138,7 +139,20 @@ def solc_exists(version):
     elif sys.version_info[1] >= 6:
         # we are using solc-x for the the 0.5 and higher
         try:
-            return solcx.install.get_executable(version)
+            if platform.system() == "Darwin":
```
Comment (norhh) [inline mythril/ethereum/util.py:142]: This part can be made a function which can be reused in the mythril_disassembler.py right?

Comment (jimtje) [reply mythril/ethereum/util.py:142]: Yeah I wasn't quite sure if that was used anywhere else, but yes effectively it's the same code twice. The code in mythril_disassembler.py throws exception if no version is found either on the system or for download

**Comment by norhh** [comment]
Looks good, but It would be good but many people specify versions as `--solc v0.5.0` , for that something like 
```
if version[0] != "v":
    version = "v"+version 
``` 
being at the beginning of the function would be good. This will also reduce the number of `"v"+ version` in the code.

**Comment by jimtje** [comment]
> Looks good, but It would be good but many people specify versions as `--solc v0.5.0` , for that something like
> 
> ```
> if version[0] != "v":
>     version = "v"+version 
> ```
> 
> being at the beginning of the function would be good. This will also reduce the number of `"v"+ version` in the code.

I think the problem with that is because I'm doing version checks with semver for compatibility, it pretty much either needs a "v" + in front of half of the times I reference version, or I have to do version[1:] for the version checking after I add the "v" to the version anyway. Both are used 3-4 times or so I think. I think that in mythril_disassembler.py since version is only compared to the version number extracted by regex, it makes sense slice the input to cut out the leading "v" if it's there. But I'm not sure it's necessarily better to use Version(version[1:]) instead of solcx.install("v" + version) in utils.py, would have to fiddle with the two requiring different forms of input in either case. 🤷‍♂️

**Review by norhh — APPROVED** [review]
Great work, thanks for the PR 🚀

**Pull request merged**